src/NHapi.Base/Parser/ParserBase.cs

```diff
@@ -369,7 +369,7 @@ protected internal virtual IMessage InstantiateMessage(string theName, string th
             var messageClass = Factory.GetMessageClass(theName, theVersion, isExplicit);
             if (messageClass == null)
             {
-                throw new Exception($"Can't find message class in current package list: {theName}");
+                throw new HL7Exception($"Can't find message class in current package list: {theName}");

```

Comment:
Does this need a specific error code? or is the default one applicable here?

Reply:
I added an UNSUPPORTED_MESSAGE_TYPE ErrorCode to this

Reply:
From what I have read about the ErrorCodes this should be the correct one in this case?